frontend/public/components/build-pipeline.tsx

```diff
@@ -0,0 +1,174 @@
+import * as React from 'react';
+import * as _ from 'lodash-es';
+import { fromNow } from './utils/datetime';
+import { K8sResourceKind } from '../module/k8s';
+
+const getBuildNumber = (resource: K8sResourceKind): number => _.get(resource, ['metadata', 'annotations', 'openshift.io/build.number']);
+const getStages = (status): any[] => (status && status.stages) || [];
+const getJenkinsStatus = (resource: K8sResourceKind) => {
+  const status = _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-status-json']);
+  if (!status) {
+    return {};
+  }
+
+  const json = _.attempt(JSON.parse, status);
+  return _.isError(json) ? {} : json;
+};
+const getJenkinsLogURL = (resource: K8sResourceKind): string => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-log-url']);
+const getJenkinsBuildURL = (resource: K8sResourceKind): string => _.get(resource, ['metadata', 'annotations', 'openshift.io/jenkins-build-uri']);
+
+const BuildSummaryStatusIcon: React.SFC<BuildSummaryStatusIconProps> = ({status}) => {
+  const statusClass = _.lowerCase(status);
+  const icon = ({
+    new: 'fa-hourglass-o',
+    pending: 'fa-hourglass-half',
+    running: 'fa-refresh fa-spin',
+    complete: 'fa-check-circle',
+    failed: 'fa-times-circle'
+  })[statusClass];
+  const hide = icon ? 'hide' : '';
```

Comment:
Why use the `hide` class if we could just use JSX to hide the node. Is there some animation we want?

Reply:
@sg00dwin there isn't a `.hide` class in the `build-pipelines.scss` file, doing a `grep -ri 'hide' public/components'` doesn't reveal a generic hide class with an animation, so I'll update this to use `jsx`.

Reply:
Will be:

```html
{!icon && <span className="build-pipeline__status-icon">
      <span className="fa fa-refresh fa-spin" aria-hidden="true"></span>
</span>}
```

Reply:
Yeah, suggest an sr-only span in there too for screen readers